app/src/main/java/com/sebastianvm/mango/data/UserStore.kt

```diff
@@ -0,0 +1,36 @@
+package com.sebastianvm.mango.data
+
+import com.sebastianvm.mango.database.MangoDatabase
+import com.sebastianvm.mango.database.models.UserEnt
+import com.sebastianvm.mango.model.User
+import com.sebastianvm.mango.util.coroutines.IODispatcher
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.android.components.ViewModelComponent
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.withContext
+
+class UserStore (
+    private val db: MangoDatabase,
+    private val ioDispatcher: CoroutineDispatcher
+) {
```

Comment:
We want to use hilt to provide the dependencies, so this should be of the format `@Inject  constructor`

Reply:
Hmm I think the nested module thing is confusing me

Reply:
figured it out